spec/controllers/spaces_spec.rb

```diff
@@ -42,13 +44,162 @@ def setup_knock(user)
+
+          context 'when space has organization' do
+            context 'when creator is member' do
+              let (:organization) {
+                organization = FactoryGirl.create(:organization)
+                FactoryGirl.create(:user_organization_membership, user: creator, organization: organization)
+                organization
+              }
+              it { is_expected.to respond_with :ok }
+            end
+
+            context 'when creator is not member' do
+              let (:organization) { FactoryGirl.create(:organization) }
+              it { is_expected.to respond_with :unauthorized }
+            end
+          end
+        end
+      end
+    end
+  end
+
+  describe 'GET index' do
+    let (:creator) { FactoryGirl.create(:user, :lite_plan) }
+    let (:secondary) { FactoryGirl.create(:user, :lite_plan) }
+    let (:organization_creator_member) {
+      organization = FactoryGirl.create(:organization, name: 'creator member')
+      FactoryGirl.create(:user_organization_membership, user: creator, organization: organization)
+      organization
+    }
+    let (:secondary_organization) { FactoryGirl.create(:organization, name: 'creator not member') }
+    let (:spaces) {{
+      creator_public_no_org: FactoryGirl.create(:space, user: creator, is_private: false, name: 'public no org'),
+      creator_private_no_org: FactoryGirl.create(:space, user: creator, is_private: true, name: 'private no org'),
+      creator_public_org: FactoryGirl.create(:space, organization: organization_creator_member, user: creator, is_private: false, name: 'public org'),
+      creator_private_org: FactoryGirl.create(:space, organization: organization_creator_member, user: creator, is_private: true, name: 'private org'),
+      creator_public_org_no_member: FactoryGirl.create(:space, organization: secondary_organization, user: creator, is_private: false, name: 'private org not member'),
+      creator_private_org_no_member: FactoryGirl.create(:space, organization: secondary_organization, user: creator, is_private: true, name: 'private org not member'),
+      secondary_public_no_org: FactoryGirl.create(:space, user: secondary, is_private: false, name: 'public no org secondary user'),
+      secondary_private_no_org: FactoryGirl.create(:space, user: secondary, is_private: true, name: 'private no org secondary user'),
+      secondary_public_org: FactoryGirl.create(:space, organization: organization_creator_member, user: secondary, is_private: false, name: 'public org secondary user'),
+      secondary_private_org: FactoryGirl.create(:space, organization: organization_creator_member, user: secondary, is_private: true, name: 'private org secondary user'),
+    }}
+
+    #TODO(matthew): I want to just parse this via the spaces representer but that seems impossible.
+    #let (:rendered_spaces) { SpacesRepresenter.new([]).from_json(response.body) }
+    let (:rendered_space_ids) { JSON.parse(response.body)['items'].map {|s| s['id']} }
```

Comment:
It's definitely possible, though may be a bit tricky.

https://github.com/apotonick/roar (see section on parsing)